maildir.go

```diff
@@ -123,6 +123,24 @@ func (d Dir) UnseenCount() (int, error) {
 	return c, nil
 }
 
+func parseKey(filename string) string {
+	split := strings.FieldsFunc(filename, func(r rune) bool {
+		return r == separator
+	})
+
+	if len(split) == 0 {
+		return ""
+	}
+
+	return split[0]
+}
+
+// Key returns the key for the given file path.
+func (d Dir) Key(path string) string {
+	filename := filepath.Base(path)
```

Comment:
Can we check `filepath.Dir(path) == string(d)` to avoid returning keys for other Maildirs?

Reply:
Good catch, thanks